src/representation_theory/monoid_algebra_basis.lean

```diff
@@ -0,0 +1,187 @@
+/-
+Copyright (c) 2022 Amelia Livingston. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Amelia Livingston
+-/
+
+import representation_theory.Rep
+import representation_theory.basic
+
+/-!
+# The structure of the `k[G]`-module `k[Gⁿ]`
+
+This file contains API for the `k[G]`-module `k[Gⁿ]`, where `k` is a commutative ring, `G` is
+a group, and we express the module structure as the representation `G →* End(k[Gⁿ])` induced by the
+diagonal action of `G` on `Gⁿ.`
+
+In particular, we define morphisms of `k`-linear `G`-representations between `k[Gⁿ⁺¹]` and
+`k[G] ⊗ₖ k[Gⁿ]` (on which `G` acts by `ρ(g₁)(g₂ ⊗ x) = (g₁ * g₂) ⊗ x`) which we will later show
+define an isomorphism. (Will finish in the morning).
```

Comment:
I want to talk about how this stuff will be used to show $k[G^{n + 1}]$ is free over $k[G]$, but I'm not sure how this should be stated - after passing to the Module category? I ultimately only need that it's projective anyway.

Reply:
Could you just state it as `module.free (monoid_algebra k G) (as_module ρ)` where as module is a type synonym for the vector space which has a `module (monoid_algebra k G)` instance? (#13713 defines such this `as_module`)